### PR TITLE
Disable delete action when user is locked (task #6800)

### PR DESCRIPTION
**Pull request by softius**
A user can be deleted only and only if:

- User is not the system user defined in .env
- User is not the currently logged in user

### CODE REVIEW

File: tests/TestCase/Controller/UsersControllerTest.php [hunk 1]
```diff
@@ -320,28 +321,45 @@ public function testEditWithoutSession()
...
+    public function testDeleteSameUser()
+    {
+        $this->enableCsrfToken();
+        $this->enableSecurityToken();
+        $this->withSession();
+
+        try {
+            $this->disableErrorHandlerMiddleware();
+            $this->delete('/users/delete/' . $this->userId);
+            $this->fail('Expected ForbiddenException');
```
Comment (mamchenkov) [inline tests/TestCase/Controller/UsersControllerTest.php]: @softius there is an easier way to test for expected exceptions.  See an example here:

https://github.com/QoboLtd/project-template-cakephp/blob/master/tests/TestCase/Feature/FactoryTest.php#L34

File: src/Template/Users/index.ctp [hunk 2]
```diff
@@ -94,16 +94,30 @@ echo $this->Html->scriptBlock(
...
+                                            'disabled' => true,
+                                            'onClick' => 'return false',
+                                        ]
+                                    ) ?>
+                                <?php else: ?>
+                                    <?= $this->Form->postLink(
+                                        '<i class="fa fa-trash"></i>',
+                                        ['controller' => 'Users', 'action' => 'delete', $user->id],
+                                        [
+                                            'confirm' => __('Are you sure you want to delete # {0}?', $user->id),
```
Comment (mamchenkov) [inline src/Template/Users/index.ctp]: @softius while we are at it, I think we should show `$user->username` in the confirmation dialog, and not the `$user->id`.  It's more humane :)

File: tests/TestCase/Controller/UsersControllerTest.php (hunk 1 above)
Comment (softius) [reply tests/TestCase/Controller/UsersControllerTest.php]: Totally forgot about this! :+1:

File: src/Template/Users/index.ctp (hunk 2 above)
Comment (softius) [reply src/Template/Users/index.ctp]: @mamchenkov We can also remove the data table search which is enabled for Users (for some reason)

Comment (mamchenkov) [reply src/Template/Users/index.ctp]: @softius there was a reason for the dataTables search, but I don't remember it top of my head.  Let's leave it for now.  Later I'll dig through the tickets and git history to see why it's there and if we need it at all. :)

File: src/Template/Users/index.ctp [hunk 3]
```diff
@@ -94,16 +94,30 @@ echo $this->Html->scriptBlock(
...
-                                    '<i class="fa fa-trash"></i>',
-                                    ['controller' => 'Users', 'action' => 'delete', $user->id],
-                                    [
-                                        'confirm' => __('Are you sure you want to delete # {0}?', $user->id),
-                                        'title' => __('Delete'),
-                                        'class' => 'btn btn-default btn-sm',
-                                        'escape' => false
-                                    ]
-                                ) ?>
+                                <?php if (in_array($user->username, $lockedUsers)): ?>
```
Comment (anvyst) [inline src/Template/Users/index.ctp]: @softius I thinks it would be better if we simply hide delete button if the user cannot delete a record.

Comment (mamchenkov) [reply src/Template/Users/index.ctp]: I think it's fine ;)

**Pull request merged**